src/control/server/config.go

```diff
@@ -444,5 +447,86 @@ func (c *Configuration) Validate(log logging.Logger) (err error) {
 			}
 		}
 	}
+
+	if len(c.Servers) > 1 {
+		if err := validateMultiServerConfig(log, c); err != nil {
```

Comment:
I thjink it would also be helpful to validate multi io-server config, similarly, to look for overlapping resources such as fabric_interface and NUMA node assignment.  Is that something you want to handle now or in a separate PR?

Reply:
It does validate fabric interface... It checks that each server has a unique tuple of (provider,fabric_interface,fabric_interface_port). Is there something else it should check?

I'm not sure we want to flag duplicate numa nodes as an error, because there may be reasons for wanting to run that way, right? In other words, it may be a suboptimal configuration but it's not an error that will cause things to fail later, will it?

Reply:
The check for the unique (provider, fabric_interface, fabric_interface_port) tuple is OK if you are not concerned with the NUMA node.  My comment could have been worded better to say "look for overlapping resources of (fabric_interface, NUMA node) tuples."  

I remember reading that the SPDK needed each primary mode instance to be initialized in its own core.  One way of achieving that is to assign unique NUMA nodes to each io-server, which match the NUMA nodes for their storage and network devices.  I just did a quick check of the SPDK documents but don't see the need for core separation explicitly stated.  Maybe I made it up.  Other than the potential SPDK issue, I am aware of no other problem other than performance degradation if there is a NUMA node conflict.

Reply:
My inclination would be to land what we have now as a first pass at catching obviously fatal config problems and then build on it as we learn more. If you are concerned about the numa node issue, then my suggestion would be for you to create a ticket where you can record the results of your research and/or add a PR to improve the validation.

Reply:
I found this: https://spdk.io/doc/nvme.html which is the closest I can find to what I remember reading. The interesting portion is this:

Limitations

1.  Two processes sharing memory may not share any cores in their core mask.
2.  If a primary process exits while secondary processes are still running, those processes will continue to run. However, a new primary process cannot be created.
3.  Applications are responsible for coordinating access to logical blocks.
4.  If a process exits unexpectedly, the allocated memory will be released when the last process exits.

It's not exactly what I remember.  I think we may need to explore #1, but it's not a blocker for now.